src/card_flows/card_flow_reconstruct_wallet.c

```diff
@@ -96,13 +96,46 @@
  * GLOBAL FUNCTIONS
  *****************************************************************************/
 card_error_type_e card_flow_reconstruct_wallet(uint8_t threshold) {
-  // TODO: Support this for threshold number of cards
+  card_error_type_e result = CARD_OPERATION_DEFAULT_INVALID;
 
-  card_fetch_share_cfg_t configuration = {.heading = ui_text_tap_1_2_cards,
-                                          .message = ui_text_place_card_below,
-                                          .skip_card_removal = true,
-                                          .xcor = 0,
-                                          .remaining_cards = 0xF};
+  card_fetch_share_configuration_t configuration = {0};
+  configuration.xcor = 0;
+  configuration.card_config.acceptable_cards = ACCEPTABLE_CARDS_ALL;
+  configuration.frontend.heading = ui_text_tap_1_2_cards;
+  configuration.frontend.msg = ui_text_place_card_below;
+  configuration.card_config.skip_card_removal = false;
 
-  return card_fetch_share(&configuration);
+  card_fetch_share_response_t response = {0};
+
+  for (uint8_t xcoord = 0; xcoord < threshold; xcoord++) {
+    // Don't accept the same card again
+    configuration.card_config.acceptable_cards ^=
+        response.card_info.tapped_card;
+
+    // Change heading for second card onwards
+    if (0 != xcoord) {
+      configuration.frontend.heading = ui_text_tap_2_2_cards;
+    }
+
+    // Skip card removal in last card
+    if (threshold == xcoord + 1) {
+      configuration.card_config.skip_card_removal = true;
+    }
+
+    configuration.xcor = xcoord;
```

Comment:
At some point keep a limit on the xcor, as buffers are already hardcoded. Better check the limit in the `card_fetch_share` api

Reply:
Added check for threshold